src/shared/components/Searchbar/Searchbar.style.tsx

```diff
@@ -39,25 +40,47 @@ export const CancelButton = styled(IconButton)`
   }
 `
 
-export const Container = styled.div`
-  position: relative;
+export const Container = styled.div<{ hasFocus: boolean; hasQuery: boolean }>`
   display: flex;
   align-items: center;
+  transition: width ${transitions.timings.regular} ${transitions.easing};
+  will-change: width;
+  top: 0;
+  right: 0;
   box-shadow: inset 0 0 0 1px ${colors.gray[700]};
+  ${({ hasFocus, hasQuery }) => `
+    height: ${hasFocus ? '64px' : '39px'};
+    position: ${hasFocus ? 'fixed' : 'relative'};
+    width: ${hasQuery || hasFocus ? '100%' : '39px'};
+    padding-left: ${hasFocus || hasQuery ? sizes(2) : 0};
+    background-color: ${hasFocus ? colors.gray[800] : 'transparent'};
+    ${
+      !hasFocus &&
+      `
+      border: 1px solid ${colors.gray[700]};
+      margin-left: auto;`
+    };
+  `};
```

Comment:
This block will be not parsable by tooling because it's just a string. Let's extract this to some function for example like this:

```tsx
const getContainerFocusStyles = ({hasFocus, hasQuery}) => css`
  height: ${hasFocus ? '64px' : '39px'};
  // ...
`

const Container = styled.div`
  ${getContainerFocusStyles};
`